### PR TITLE
fix: `explorer-desktop` not closing when the `kernel communication` is closed

### PR DESCRIPTION
**Pull request by Kinerius**
## What does this PR change?

The application will close if the WebSocket connection is closed, this was removed because we needed to close the connection in order for a restart to work. Now, this functionality is back and is compatible with the restart functionality.

## How to test the changes?

Close the client and the application should close.
Restart the client if there is a timeout and the application shouldn't close.

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

File: unity-renderer-desktop/Assets/Scripts/MainScripts/DCL/MainDesktop/MainDesktop.cs [hunk 1]
```diff
@@ -51,6 +71,7 @@ protected override void Start()
 
         private void Reload()
         {
+            isRestarting = true;
             kernelCommunication.Dispose();
             SceneManager.LoadScene(0);
```
Comment (BrianAmadori) [inline unity-renderer-desktop/Assets/Scripts/MainScripts/DCL/MainDesktop/MainDesktop.cs:76]: I'm wondering about this line, why do we matter to set `isRestarting` to true if we are going to wipe the scene? Wiping the scene should remove this very `MonoBehaviour`.

Comment (Kinerius) [reply unity-renderer-desktop/Assets/Scripts/MainScripts/DCL/MainDesktop/MainDesktop.cs:76]: When the WebSocket communication is disposed, we catch that event and close the application in this class, we don't want that to happen when reloading I used this flag.

**Review by kuruk-mm — APPROVED** [review]
Great! Try to edit the commit message to refer to each part for the component name.

File: unity-renderer-desktop/Assets/Scripts/MainScripts/DCL/MainDesktop/MainDesktop.cs (hunk 1 above)
Comment (BrianAmadori) [reply unity-renderer-desktop/Assets/Scripts/MainScripts/DCL/MainDesktop/MainDesktop.cs:76]: Ok, got it.

**Review by BrianAmadori — APPROVED** [review]

**Pull request merged**